feat: allow to disable rules (#2743)
Mergifyio/mergify-engine: docs/source/configuration.rst

     - string
     - The name of the rule. This is not used by the engine directly, but is
       used when reporting information about a rule.
   * - ``disabled``
     - dictionary with ``reason`` key

>      - if set this disable the rule and cancel any ongoing actions. A reason
>        string must be set

Reviewer: ```suggestion
     - This optional key allows to disabled a rule and cancel any ongoing actions. A reason
       must be set using the ``reason`` key.
```

You need to add an example.

          merge:
            method: merge

> Disabling rule

Reviewer: ```suggestion
Disabling Rules
```

~~~~~~~~~~~~~~

> You can disable a rule but keeping it in the configuration. This allows to
> gracefully handle the cancellation of any ongoing actions (like stopping the
> merge queue).

Reviewer: ```suggestion
You can disable a rule while keeping it in the configuration. This allows
gracefully handling the cancellation of any ongoing actions (e.g., stopping the
merge queue).
```

> You can disable a rule while keeping it in the configuration. This allows
> gracefully handling the cancellation of any ongoing actions (e.g., like stopping
> the merge queue).

Reviewer: ```suggestion
You can disable a rule while keeping it in the configuration. This allows
gracefully handling the cancellation of any ongoing actions (e.g., stopping
the merge queue).
```
